### ISSUE TITLE
[FEATURE REQUEST] Move sort menu from toolbar to files view

### ISSUE THREAD

**Issue by abelgardep** [owncloud/android#2848]
Continuation of #2742 

Sort menu will be visible via modal bottom sheet where you can select to sort alphabetically, by date or size. You will be able to order ascending or descending.

You will be able to swich between grid and list view. First mockup:

![sort_above_files](https://user-images.githubusercontent.com/47524927/79346754-05778500-7f33-11ea-9cb2-2760c4f33a86.png)

## AC

- Menu is scrollable, always above the top item on the list, left side. Will include the arrow to show asc/desc order
- Include the grid/list at the right
- Wording: Name, Date, Type, Size
- Arrow to show ascending/descending will only be display on the selected option of the bottom sheet
- Selected option will be highlighted
- By clicking on the selected option, it will switch the order (ascending/descending)
- Size and Type Icons are still in consideration
- Once selection is done, bottom sheet will be closed, top bar updated, and list correctly sorted.

![image](https://user-images.githubusercontent.com/60343863/98016536-65b21380-1dfe-11eb-84eb-88ec50ec2cce.png)

2. Figma Mockup
https://www.figma.com/file/EVQQcpITjVPiF4yT3eWKuf/Android-Proposals?node-id=0%3A1

### TASKS

- [x] Research (if needed)
- [x] Create branch new_design/new_sort_menu
- [x] Development tasks
  - [x] Create a new sort menu in file list fragment
  - [x] Create a button to switch between grid and list mode
  - [x] Delete old way to handle grid and list changes
  - [x] Create a bottom sheet dialog fragment to choose how to sort the filelist
  - [x] Some accurate design (ask for someone in the company)
  - [x] Replace also the sort menu from ReceiveExternalFilesActivity
  - [x] Delete old way to handle file list sorting
  - [x] Hide sorting menu if multi-selection mode is active
  - [x] Add a button to handle ascending or descending
  - [x] Delete old way to set asc or desc
  - [ ] Add rounded corners to bottom sheet fragment [TODO in next iterations]
  - [x] Apply same style for every bottom sheet fragment
  - [x] Decide behaviour when receiving a file from another app
  - [x] Show create folder instead of grid/switch option when receiving files
  - [ ] Make it scrollable [TODO in next iterations]
  - [ ] Add UI tests [TODO in next iterations]
- [x] Code review and apply changes requested
- [ ] Design test plan
- [ ] QA
- [ ] Merge branch new_design/new_sort_menu into master

### PR

- App: #3012

**Comment by abelgardep** [comment]
### AC to discuss

1. `Select all` was selectable without entering multiselection mode. From now on, it will be available only on multiselection menu
2. What happens with `Refresh account` if we remove that menu from toolbar?
3. Will sort menu be fixed or scrollable?

**Comment by jesmrec** [comment]
> 1. Select all was selectable without entering multiselection mode. From now on, it will be available only on multiselection menu

WFM

> 2. What happens with Refresh account if we remove that menu from toolbar?

We can do it in the moment the account is attached. But, without the option, only in that moment is posible to do. Is it posible to add any other button just next to the avatar or the grid/list?

> 3. Will sort menu be fixed or scrollable?

I like the fixed option more that scrollable. Just for giving you a random example, the oC iOS app implements the scrollable option.

**Comment by CarolinaCst** [comment]
Additional changes

![image](https://user-images.githubusercontent.com/60343863/96723688-266fd580-13af-11eb-849f-2da378b0bcc4.png)

![image](https://user-images.githubusercontent.com/60343863/96723711-325b9780-13af-11eb-84a6-db40fc82d12d.png)

**Comment by CarolinaCst** [comment]
After the Meeting of 03.11.2020, here are the following agreements:

1. Sort menu

- Scrollable

- Wording new: Name, Date, Type, Size

- Arrow will only be display on the selected option 

- Selected option will be highlighted 

- Size and Type Icons are still in consideration

![image](https://user-images.githubusercontent.com/60343863/98016536-65b21380-1dfe-11eb-84eb-88ec50ec2cce.png)

2. Figma Mockup
https://www.figma.com/file/EVQQcpITjVPiF4yT3eWKuf/Android-Proposals?node-id=0%3A1

**Comment by jesmrec** [comment]
AC copied to the first message on the top

**Comment by jesmrec** [comment]
One additional idea: this should be included somehow in the picker view for the `copy` and `move` options, as well as the picker in the share sheet. Current o new issue @abelgardep @CarolinaCst?

**Comment by abelgardep** [comment]
> One additional idea: this should be included somehow in the picker view for the `copy` and `move` options, as well as the picker in the share sheet and the camera upload target selector. Current o new issue @abelgardep @CarolinaCst?

In the picker view for `copy` and `move` and `camera upload`, the behaviour should be the same, shouldn't it? BTW, i think we should do related stuff in the current issue.

I have some questions:
1. When receiving files from other apps, should we keep the same behaviour? Currently there is no option to switch between grid and list in that view. Maybe we could introduce create folder in that place
2. Which should be the behaviour when we switch to multi-selection mode? Should the sort options disappear?

**Comment by CarolinaCst** [comment]
> One additional idea: this should be included somehow in the picker view for the `copy` and `move` options, as well as the picker in the share sheet and the camera upload target selector. Current o new issue @abelgardep @CarolinaCst?

- Yes, agree to remain on the current issue

- Good idea, to keep the same behaviour in other sections

- Multi-selection mode should be scrollable as well

**Comment by theScrabi** [comment]
Please update branch then move it to QA pending.

**Comment by theScrabi** [comment]
Ok this is a bit to late to bring it up in this issue right now, but I thought this was interesting never the less. So maybe this could be made into an issue sometime in the future. So from what I learned in UI classes at university is that it is a good idea to reduce cursor/finger travel. What we have done with this issue is something where the user must click on the top and then let his thumb travel all the way to the bottom to select an option. This might be annoying after some time. Maybe it was a better way to put the  menu where the thumb is already resting. I've seen that rocket chat does something like this, so maybe this is something to consider:

[![cut mp4](https://user-images.githubusercontent.com/1891273/99067664-62134f00-25ab-11eb-92ba-7136f71d5fbc.gif)](https://imgur.com/a/WyxNxui)

**Comment by michaelstingl** [comment]
> Maybe it was a better way to put the menu where the thumb is already resting.

Very good point! @CarolinaCst what does the Material Design guidelines say? Examples from other apps?
